### PR TITLE
Fix downloading of signed files

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla-jetpack/jpm/issues/376

### CODE REVIEW

File: test/unit/test.amo-client.js [hunk 1]
```diff
@@ -328,18 +328,57 @@ describe('amoClient.Client', function() {
...
+      expect(conf.headers).to.have.keys(['Accept', 'Authorization']);
+    });
+
+    it('preserves request headers', function() {
+      var headers = {'X-Custom': 'thing'};
+      var conf = this.client.configureRequest({
+        url: '/path',
+        headers: headers,
+      });
+      expect(conf.headers['X-Custom']).to.be.equal(headers['X-Custom']);
```
Comment (mstriemer) [inline test/unit/test.amo-client.js]: I'd rather see: `to.be.equal('thing')`

File: test/unit/test.amo-client.js [hunk 2]
```diff
@@ -328,18 +328,57 @@ describe('amoClient.Client', function() {
...
+    });
+
+    it('allows you to override request headers', function() {
+      var headers = {'Accept': 'text/html'};
+      var conf = this.client.configureRequest({
+        url: '/path',
+        headers: headers,
+      });
+      expect(conf.headers['Accept']).to.be.equal(headers['Accept']);
```
Comment (mstriemer) [inline test/unit/test.amo-client.js]: Same here. Wouldn't be too hard for some typos to make this `undefined === undefined`

File: test/unit/test.amo-client.js [hunk 3]
```diff
@@ -328,18 +328,57 @@ describe('amoClient.Client', function() {
...
+    });
+
+    it('makes relative URLs absolute', function() {
+      var urlPath = '/somewhere';
+      var conf = this.client.configureRequest({url: urlPath});
+      expect(conf.url).to.be.equal(this.client.apiUrlPrefix + urlPath);
```
Comment (mstriemer) [inline test/unit/test.amo-client.js]: This should either check the value of `this.client.apiUrlPrefix` or set it in the test.

**Comment by mstriemer** [comment]
r+wc

File: test/unit/test.amo-client.js (hunk 1 above)
Comment (kumar303) [inline test/unit/test.amo-client.js]: why? then `thing` would have to be updated in two places

Comment (kumar303) [inline test/unit/test.amo-client.js]: just saw your rationale about `undefined`. Ugh. Good point.

Comment (mstriemer) [inline test/unit/test.amo-client.js]: You're never going to update it so it doesn't matter. And find/replace is going to sort that out in a matter of seconds.

It's more clear. When someone looks at this they don't have to look back to verify that `headers['X-Custom']` actually has a meaningful value and there's no chance of typoing the key and having the test stop working.

File: test/unit/test.amo-client.js (hunk 3 above)
Comment (kumar303) [inline test/unit/test.amo-client.js]: I moved it to a test instance var which should add some protection against `undefined` https://github.com/kumar303/jpm/blob/fix-download/test/unit/test.amo-client.js#L20

**Pull request merged**